WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostDetailFragment.kt

```diff
@@ -371,26 +360,42 @@ class ReaderPostDetailFragment : ViewPagerFragment(),
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        initViewModel()
+        initViewModel(savedInstanceState)
 
         showPost()
     }
 
-    private fun initViewModel() {
+    private fun initViewModel(savedInstanceState: Bundle?) {
         viewModel = ViewModelProvider(this, viewModelFactory).get(ReaderPostDetailViewModel::class.java)
 
-        viewModel.uiState.observe(viewLifecycleOwner, { renderUiState(it) })
+        viewModel.uiState.observe(viewLifecycleOwner, {
+            uiHelpers.updateVisibility(text_error, it.errorVisible)
+            uiHelpers.updateVisibility(progress_loading, it.loadingVisible)
+            when (it) {
+                is LoadingUiState -> Unit // Do Nothing
+
+                is ReaderPostDetailsUiState -> renderUiState(it)
+
+                is ErrorUiState -> {
+                    uiHelpers.updateVisibility(signInButton, it.signInButtonVisibility)
+                    val message = it.message?.let { msg -> uiHelpers.getTextOfUiString(requireContext(), msg) }
+                    showError(message?.toString())
+                }
+            }
+        })
 
         viewModel.refreshPost.observe(viewLifecycleOwner, {} /* Do nothing */)
 
         viewModel.snackbarEvents.observe(viewLifecycleOwner, { it?.applyIfNotHandled { showSnackbar() } })
 
         viewModel.navigationEvents.observe(viewLifecycleOwner, { it.applyIfNotHandled { handleNavigationEvent() } })
 
-        viewModel.start(isRelatedPost)
+        viewModel.start(isRelatedPost, savedInstanceState ?: arguments)
```

Comment:
Suggestion (💡): Instead of passing a `bundle`, which is Android specific, maybe passing the individual `isFeed` and `interceptedUri` arguments is better? I know that this is already happening and we are used to passing a `bundle` to the view model, but given the fact that view models should be framework agnostic maybe it is better if we separate those?

PS: I know that this is an `ViewModel` specific to Android, but still the general suggestion is to avoid putting any Android specific code in it.

Reply:
I twice thought about it during refactoring, and noticed this existing convention so passed bundle for consistency. 
I agree it is better to avoid it, suggestion accepted in a959c8a.

Reply:
🙏